common/meta/raft-store/src/state_machine/database_lookup.rs

```diff
@@ -0,0 +1,99 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+use std::io::Cursor;
+
+use bytes::BytesMut;
+use common_exception::ErrorCode;
+use common_io::prelude::BinaryRead;
+use common_io::prelude::BinaryWriteBuf;
+use common_meta_sled_store::sled::IVec;
+use common_meta_sled_store::SledOrderedSerde;
+use serde::Deserialize;
+use serde::Serialize;
+
+const DB_LOOKUP_KEY_DELIMITER: u8 = b'/';
+
+#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
+pub struct DatabaseLookupKey {
+    tenant: String,
+    delimiter: u8,
+    database_name: String,
+}
+
+impl DatabaseLookupKey {
+    pub fn new(tenant: String, database_name: String) -> Self {
+        DatabaseLookupKey {
+            tenant,
+            delimiter: DB_LOOKUP_KEY_DELIMITER,
+            database_name,
+        }
+    }
+
+    pub fn get_database_name(&self) -> String {
+        self.database_name.clone()
+    }
+}
+
+impl SledOrderedSerde for DatabaseLookupKey {
+    fn ser(&self) -> Result<IVec, ErrorCode> {
+        let mut buf = BytesMut::new();
+
+        if buf.write_string(&self.tenant).is_ok()
+            && buf.write_scalar(&self.delimiter).is_ok()
+            && buf.write_string(&self.database_name).is_ok()
+        {
+            return Ok(IVec::from(buf.to_vec()));
+        }
+        Err(ErrorCode::MetaStoreDamaged("invalid key IVec"))
```

Comment:
It's not meta-store's fault...

BTW, IMHO, an error should include the original cause in it.
In our case, the reason for which it failed to serialize tenant or database name.